code/modules/mod/mod_activation.dm

```diff
@@ -76,6 +76,8 @@
 			RegisterSignal(part, COMSIG_ATOM_EXITED, .proc/on_overslot_exit)
 	if(wearer.equip_to_slot_if_possible(part, part.slot_flags, qdel_on_fail = FALSE, disable_warning = TRUE))
 		ADD_TRAIT(part, TRAIT_NODROP, MOD_TRAIT)
+		if(istype(part, /obj/item/clothing/shoes/mod)) //if the user has digitigrade legs, we need to redraw them when the boots are deployed
+			wearer.update_mutant_bodyparts()
```

Comment:
this is really not clean. why not just put update_mutant_bodyparts on the code that handles foot icons updating? suits/jumpsuits already do it

Reply:
Thanks for the tip, that makes a lot more sense. I've made that change, and tested to confirm that it works the same way.